### ISSUE TITLE
how use this code?

### ISSUE THREAD

**Issue by ThatNight** [pdpino/medical-ai#1]
Hi, this code helped me a lot. May I ask how to use this code? How can the dataset/ui_xray code run successfully? What should I do if I want to use chexpert to extract the labels of the iu_xray dataset? Looking forward to your reply

**Comment by pdpino** [comment]
Hi @ThatNight !

> I want to use chexpert to extract the labels of the iu_xray dataset

* I'd say the easiest way is using directly the original tool: https://github.com/stanfordmlgroup/chexpert-labeler. The authors provide a script to label reports, i.e. `python label.py <reports>`
* For convenience I wrapped it in python like this: https://github.com/pdpino/medical-ai/blob/181153329e9d6d58a51a27fdbfdadc729d9de9bb/medai/metrics/report_generation/chexpert.py#L150

---

> How can the dataset/ui_xray code run successfully

* If you want to use the code directly: use the class [`IUXRayDataset`](https://github.com/pdpino/medical-ai/blob/ee87012366335f2212c1f9babeeb1f43244d53f7/medai/datasets/iu_xray.py#L42), you should setup a folder like the example provided in [`workspace/iu-examples`](https://github.com/pdpino/medical-ai/tree/master/workspace/iu-examples): with the images, clean reports in JSON and vocabulary. Set the environment variable `DATASET_DIR_IU_XRAY` to point to that folder, e.g. `export DATASET_DIR_IU_XRAY=/path/to/data/iu-examples`
* I typically used this function to create a pytorch Dataloader: https://github.com/pdpino/medical-ai/blob/ee87012366335f2212c1f9babeeb1f43244d53f7/medai/datasets/__init__.py#L237
* Main requirements are torch, torchvision, opencv, pandas, numpy. See [requirements](https://github.com/pdpino/medical-ai/blob/master/requirements.txt) for the full list (though some library versions might be somewhat old by now)


Tbh, the code grew a lot and there is a lot of additional stuff for running experiments, debugging and other things. You can probably delete some methods that won't be useful, or even start from scratch and use this only for reference :smile:
